lalrpop/src/lr1/codegen/base.rs

```diff
@@ -148,7 +148,7 @@ impl<'codegen, 'grammar, W: Write, C> CodeGenerator<'codegen, 'grammar, W, C> {
         if self.grammar.intern_token.is_some() {
             // if we are generating the tokenizer, create a matcher as our input iterator
             rust!(self.out,
-                  "let mut {}tokens = {}::{}intern_token::{}Matcher::new(input);",
+                  "let mut {}tokens = {}::{}intern_token::{}Matcher::new(input).peekable();",
```

Comment:
Needed to make the tokens iterator peekable so that error recovery can check if parsing the next token would succeed. Might be able to rewrite how  `tokens.next()` is done so that `peekable` is not necessary.